docroot/modules/custom/mass_content/src/Commands/MassContentCommands.php

```diff
@@ -126,4 +126,95 @@ public function migrateDateFields(string $type = '', int $limit = 0) {
+   *
+   * @command mass-content:migrate-service
+   *
+   * @usage mass-content:migrate-service 1000
+   *   1000 is the number of nodes that will be processed.
+   */
+  public function migrateServiceData(int $limit = 0) {
+    $fields = [
+      'field_service_ref_actions_2',
+      'field_link_group',
+      'field_service_ref_actions',
+      'field_service_key_info_links_6',
+      'field_template',
+      'field_service_ref_locations'
+    ];
+    // 1. Log the start of the script.
+    $this->loggerChannelFactory->get('mass_content')->info('Update nodes batch operations start');
+
+    // 2. Retrieve all nodes of this type.
+    $storage = $this->entityTypeManager->getStorage('node');
+    try {
+      $query = $storage->getQuery();
+      $query->condition('type', 'service_page');
+      $orCondition = $query->orConditionGroup();
+      foreach ($fields as $field) {
+        $and = $query->andConditionGroup();
+        $and->exists($field);
+        $orCondition->condition($and);
+      }
+      $query->condition($orCondition);
+      $query->sort('nid');
+      if ($limit !== 0) {
+        $query->range(0, $limit);
+      }
+
+      $nids = $query->execute();
+    }
+    catch (\Exception $e) {
+      $this->output()->writeln($e);
+      $this->loggerChannelFactory->get('mass_content')->error('Error found @e', ['@e' => $e]);
+    }
+    // 3. Create the operations array for the batch.
+    $operations = [];
+    $numOperations = 0;
+    $batchId = 1;
+    if (!empty($nids)) {
+      $this->output()->writeln("Preparing batches for " . count($nids) . " nodes.");
+      foreach ($nids as $nid) {
+        // Prepare the operation. Here we could do other operations on nodes.
+        $this->output()->writeln("Preparing batch: " . $batchId);
+        $operations[] = [
+          '\Drupal\mass_content\MassContentBatchManager::processServiceNode',
+          [
+            $batchId,
+            $storage->load($nid),
+            t('Updating node @nid', ['@nid' => $nid]),
+          ],
+        ];
+        $batchId++;
+        $numOperations++;
+      }
+    }
+    else {
+      $this->logger()->warning('No nodes of this type @type', ['@type' => 'service_page']);
+    }
+    // 4. Create the batch.
+    $batch = [
+      'title' => t('Updating @num node(s)', ['@num' => $numOperations]),
+      'operations' => $operations,
+      'finished' => '\Drupal\mass_content\MassContentBatchManager::processNodeFinished',
+    ];
+    // 5. Add batch operations as new batch sets.
+    batch_set($batch);
+    // 6. Process the batch sets.
+    drush_backend_batch_process();
+    // 6. Show some information.
+    $this->logger()->notice("Batch operations end.");
+    // 7. Log some information.
+    $this->loggerChannelFactory->get('mass_content')->info('Update batch operations end.');
+    \Drupal::getContainer()->get('plugin.cache_clearer')->clearCachedDefinitions();
+
+    // Turn on entity_hierarchy writes after processing the item.
```

Comment:
This comment is wrong. The code is disabling, not enabling. Also, does this need to be done for every item? Note that there is a finished callback for batches where you can do cleanup work like enabling writes.

Reply:
Code is correct, it is enabling entity hierarchy. I will look into the other part and refactor accordingly. Thank you